src/html/HTMLWriter.jl

```diff
@@ -1941,6 +1952,22 @@ function format_units(size)
     return string(round(size, digits = 2), " (", unit, ")")
 end
 
+function check_search_index_size(path, settings)
+    file_size = filesize(path)
+    file_size_format_results = format_units(file_size)
+    size_threshold_warn_format_results = format_units(settings.search_size_threshold_warn)
+
+    size_threshold_msg(var::Symbol) = """
+    Generated search index over $(var) limit:
+        Generated file size: $(file_size_format_results)
+        search_size_threshold_warn: $(size_threshold_warn_format_results)
+        Search index file:   $(path)"""
```

Comment:
This could all be within the `if` in principle. Also `size_threshold_msg` is not reused, so no need to make it a function.